lib/index.js

```diff
@@ -80,6 +86,13 @@ Optimizely.prototype.track = function(track) {
   var revenue = track.revenue();
   if (revenue) payload.revenue = revenue *= 100;
   push(['trackEvent', track.event(), payload]);
+
+  var optimizelyClientInstance = window.optimizelyClientInstance;
+  if (optimizelyClientInstance && optimizelyClientInstance.track) {
+    var attributes = track.properties();
+    var userId = attributes.userId;
```

Comment:
Gotcha -- so if you _only_ want events with `userId`, I would just add a `if` guard above and no-op if it is undefined. 

```
if (track.userId()) {

...send to full stack...
}

And we will add this requirement to our docs. We wouldn't recommend people call `.identify()` because normally a track event is intentionally missing a userId, mainly because they don't have that user's `userId` at that point.

Does that sound good? Basically don't send the event if it is an anonymous event.